capsules/src/nrf51822_serialization.rs

```diff
@@ -244,7 +278,7 @@ impl uart::ReceiveClient for Nrf51822Serialization<'_> {
 
         self.active_app.map(|appid| {
             let _ = self.apps.enter(*appid, |app, _| {
-                app.rx_buffer = app.rx_buffer.take().map(|mut rb| {
+                app.rx_buffer.mut_map(|rb| {
```

Comment:
I think you do want a `mut_map_or` here, because you want to note the case of when there wasn't a receive buffer, and when there is, how much was copied. In particular, you should be signaling `max_len`, not `rx_len` in the callback. So you need to pass this out of the `mut_map_or`.